stl/inc/flat_map

```diff
@@ -33,9 +33,10 @@ _STL_DISABLE_CLANG_WARNINGS
 
 _STD_BEGIN
 
-template <class Key, class T, class Compare = less<Key>, class KeyContainer = vector<Key>,
-    class MappedContainer = vector<T>>
-    requires same_as<Key, typename KeyContainer::value_type> && same_as<T, typename MappedContainer::value_type>
+template <class _FlatMap_Key, class _FlatMap_T, class _FlatMap_Compare = less<_FlatMap_Key>,
+    class _FlatMap_KeyContainer = vector<_FlatMap_Key>, class _FlatMap_MappedContainer = vector<_FlatMap_T>>
+    requires same_as<_FlatMap_Key, typename _FlatMap_KeyContainer::value_type>
+          && same_as<_FlatMap_T, typename _FlatMap_MappedContainer::value_type>

```

Comment:
I don't see the point of the `_FlatMap` prefix. Can we changes the names to `_Key`, `_Mapped`, `_Compare`, `_KeyContainer`, and `_MappedContainer`? (ditto below)

Reply:
Of course. I need to resolve some naming conflicts though (e.g. the friend declaration in `_Flat_map_iterator_Impl`)